### PR TITLE
Testing and refactoring

### PR DESCRIPTION
**Pull request by cojoj**
I've created another two **test classes** and also made some refactoring where I found it important.

I've also documented some methods so please check if this style of documenting is suitable - it follows the new Swift 2 standards.

### CODE REVIEW

File: XcodeServerSDKTests/HTTPUtilsTests.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    // MARK: HTTP extension tests
+    func testStringForQuery() {
+        let query = [
+            "Key1": "Value1",
+            "Key2": "Value2",
+            "Key3": "Value3"
+        ]
+        
+        XCTAssertEqual(HTTP.stringForQuery(query), "?Key3=Value3&Key1=Value1&Key2=Value2")
```
Comment (czechboy0) [inline XcodeServerSDKTests/HTTPUtilsTests.swift]: Why is the order 3..1..2?

Comment (cojoj) [inline XcodeServerSDKTests/HTTPUtilsTests.swift]: From _The Swift Programming Language_:

> Swift’s `Dictionary` type does not have a defined ordering. To iterate over the keys or values of a dictionary in a specific order, use the `sort()` method on its `keys` or `values` property.

And I just copied it from Playground, so that's why the expectation is _strange_

File: XcodeServerSDK/Utils/HTTPUtils.swift [hunk 2]
```diff
@@ -106,21 +106,23 @@ extension HTTP {
...
-                }
-                let full = "?" + "&".join(pairs)
-                return full
-            }
+        guard let query = query where query.count > 0 else {
+            return ""
         }
-        return ""
+        
+        let pairs = query.keys.map { "\($0)=\(query[$0]!)" }
```
Comment (czechboy0) [inline XcodeServerSDK/Utils/HTTPUtils.swift]: Let's do `query.keys.sort().map...` to always get the same result (which we can test properly below).

File: XcodeServerSDKTests/HTTPUtilsTests.swift (hunk 1 above)
Comment (czechboy0) [inline XcodeServerSDKTests/HTTPUtilsTests.swift]: Sure, let's sort them (I added a comment above).

File: Playgrounds/XcodeServerSDK.playground/timeline.xctimeline [hunk 3]
```diff
@@ -3,7 +3,7 @@
    version = "3.0">
    <TimelineItems>
       <LoggerValueHistoryTimelineItem
-         documentLocation = "#CharacterRangeLen=11&amp;CharacterRangeLoc=338&amp;EndingColumnNumber=18&amp;EndingLineNumber=13&amp;StartingColumnNumber=5&amp;StartingLineNumber=13&amp;Timestamp=456581904.255022"
+         documentLocation = "#CharacterRangeLen=0&amp;CharacterRangeLoc=258&amp;EndingColumnNumber=18&amp;EndingLineNumber=23&amp;StartingColumnNumber=5&amp;StartingLineNumber=23&amp;Timestamp=456844329.662716"
```
Comment (czechboy0) [inline Playgrounds/XcodeServerSDK.playground/timeline.xctimeline:6]: Can you just revert this change?

**Comment by czechboy0** [comment]
Please just revert the playground timeline xml change and I'll merge this, @cojoj. Thanks for the work you put into it!

**Comment by cojoj** [comment]
@czechboy0 I've rebased and reverted `timeline.xml` as you've asked for 😉

**Comment by czechboy0** [comment]
Thanks, @cojoj!
